.github/workflows/build.yml

```diff
@@ -26,3 +26,16 @@ jobs:
       - uses: tauri-apps/tauri-action@v0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set OS env var
+        run: echo "OS=$RUNNER_OS" >> $GITHUB_ENV
+
+      - name: Run e2e
+        run: pnpm run e2e
+
+      - name: Upload e2e artifacts
+        uses: actions/upload-artifact@v3
+        if: ${{ failure() || success() }}
```

Comment:
Are we uploading artifacts on failure and success both? If yes, then why?
And if we want to do in all cases, just use `always()`

Reply:
Oh yeah, you are right, I can change it to always(), that would be simpler.
But yeah, we only generate videos for failed tests, so if there are none of them, there would be no artifacts. In that case I think just having failure should suffice. Will change to that...